pkg/instancetype/upgrade_test.go

```diff
@@ -0,0 +1,412 @@
+			Expect(newCR).To(Equal(cr))
+
+		},
+			Entry("VirtualMachineInstancetype",
+				func() (*appsv1.ControllerRevision, error) {
+					return CreateControllerRevision(vm,
+						&instancetypev1beta1.VirtualMachineInstancetype{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "clusterinstancetype",
+							},
+							Spec: instancetypev1beta1.VirtualMachineInstancetypeSpec{
+								CPU: instancetypev1beta1.CPUInstancetype{
+									Guest: uint32(1),
+								},
+								Memory: instancetypev1beta1.MemoryInstancetype{
+									Guest: resource.MustParse("128Mi"),
+								},
+							},
+						},
+					)
+				},
+			),
+			Entry("VirtualMachineClusterInstancetype",
+				func() (*appsv1.ControllerRevision, error) {
+					return CreateControllerRevision(vm,
+						&instancetypev1beta1.VirtualMachineClusterInstancetype{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "clusterinstancetype",
+							},
+							Spec: instancetypev1beta1.VirtualMachineInstancetypeSpec{
+								CPU: instancetypev1beta1.CPUInstancetype{
+									Guest: uint32(1),
+								},
+								Memory: instancetypev1beta1.MemoryInstancetype{
+									Guest: resource.MustParse("128Mi"),
+								},
+							},
+						},
+					)
+				},
+			),
+
+			Entry("VirtualMachinePreference",
+				func() (*appsv1.ControllerRevision, error) {
+					cpuPreference := instancetypev1beta1.PreferSockets
+					return CreateControllerRevision(vm,
+						&instancetypev1beta1.VirtualMachinePreference{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "preference",
+							},
+							Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
+								CPU: &instancetypev1beta1.CPUPreferences{
+									PreferredCPUTopology: &cpuPreference,
+								},
+							},
+						},
+					)
+				},
+			),
+			Entry("VirtualMachineClusterPreference",
+				func() (*appsv1.ControllerRevision, error) {
+					cpuPreference := instancetypev1beta1.PreferSockets
+					return CreateControllerRevision(vm,
+						&instancetypev1beta1.VirtualMachineClusterPreference{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "clusterpreference",
+							},
+							Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
+								CPU: &instancetypev1beta1.CPUPreferences{
+									PreferredCPUTopology: &cpuPreference,
+								},
+							},
+						},
+					)
+				},
+			),
+		)
+
+		expectControllerRevisionCreation := func() {
+			k8sClient.Fake.PrependReactor("create", "controllerrevisions", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				created, ok := action.(testing.CreateAction)
+				Expect(ok).To(BeTrue())
+
+				createdObj := created.GetObject()
+				createdCR, ok := createdObj.(*appsv1.ControllerRevision)
+				Expect(ok).To(BeTrue())
+
+				assertObjectIsLatest(createdCR)
+
+				return true, createdObj, nil
+			})
+		}
+
+		expectVirtualMachineRevisionNamePatch := func() {
+			// TODO(lyarwood) - assert the revisionName patch, GenerateRevisionNamePatch needs to be refactored first
```

Comment:
At the time I wrote this I think I was using `GenerateRevisionNamePatch` in `upgrade.go` but I've since switched to `GenerateTestReplacePatch` so we should be able to handle that here.